core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java

```diff
@@ -33,7 +33,7 @@
 public final class DefaultSdkAutoConstructList<T> implements SdkAutoConstructList<T> {
     private static final DefaultSdkAutoConstructList INSTANCE = new DefaultSdkAutoConstructList();
 
-    private final List impl = Collections.unmodifiableList(Collections.emptyList());
+    private final List impl = Collections.emptyList();
```

Comment:
Changed this because `emptyMap`/`emptyList` are already immutable so wrapping in `unmodifiable` is redundant.